src/server/wesnothd/game.cpp

```diff
@@ -882,7 +884,8 @@ void game::process_message(simple_wml::document& data, player_iterator user)
 
 	const simple_wml::string_span& msg = (*message)["message"];
 	chat_message::truncate_message(msg, *message);
-
+	// Buffer message for sending message history to newly joined players
+	msg_queue_.push_back(std::pair<std::string, std::string>(user->info().name(), msg.to_string()));
```

Comment:
Why not save the wml? You could just use the existing history data structure and functions.

Reply:
At the `simple_wml::document& data` could be some other data that's not meant for others, so I would need to extract the message part in either case.

Later in the resend logic I will need to repackage the data structures to a `simple_wml::document` for the `server.send_to_player()` function anyway, so saving `simple_wml::string_span` doesn't seem worth the overhead.

Reply:
Or do you mean another existing data structure I have overlooked?

Reply:
You can use `record_data` and `send_history` before start for this similar to how they're used after start for the replay data.

There is no support for merging of unrelated documents. If a custom client does that then neither wesnoth nor wesnothd will understand it.